apps/examples/rtc_alarm/rtc_alarm_main.c

```diff
@@ -0,0 +1,353 @@
+
+		printf("alarm_daemon started\n");
+		usleep(500 * 1000L);
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: show_usage
+ ****************************************************************************/
+
+static void show_usage(FAR const char *progname)
+{
+	fprintf(stderr, "USAGE:\n");
+	fprintf(stderr, "\t%s [-a <alarmid>] [-r <alarmid>] [-cr] [<seconds>]\n", progname);
+	fprintf(stderr, "Where:\n");
+	fprintf(stderr, "\t-a <alarmid> set alarm by absolute time\n");
+	fprintf(stderr, "\t\t<alarmid> selects the alarm: 0..%d (default: 0)\n", CONFIG_RTC_NALARMS - 1);
+	fprintf(stderr, "\t-r <alarmid> set alarm by relative time\n");
+	fprintf(stderr, "\t\t<alarmid> selects the alarm: 0..%d (default: 0)\n", CONFIG_RTC_NALARMS - 1);
+	fprintf(stderr, "\t-c\tCancel previously set alarm\n");
+	fprintf(stderr, "\t<seconds>\n");
+	fprintf(stderr, "\t\tThe number of seconds until the alarm expires.\n");
+	fprintf(stderr, "\t\t(only if no -c option given.)\n");
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * alarm_main
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int rtc_alarm_main(int argc, char *argv[])
+#endif
+{
+	unsigned long seconds = 0;
+	bool badarg = false;
+	bool cancelmode = false;
+	bool setmode;
+	bool relset = false;
+	int opt;
+	int alarmid = 0;
+	int fd;
+	int ret;
+
+	/* Make sure that the alarm daemon is running */
+
+	ret = start_daemon();
+	if (ret < 0) {
+		return EXIT_FAILURE;
+	}
+
+	/* Parse commandline parameters. NOTE: getopt() is not thread safe nor
+	 * re-entrant.
+	 * To keep its state proper for the next usage, it is necessary to parse to
+	 * the end of the line even if an error occurs.  If an error occurs, this
+	 * logic just sets 'badarg' and continues.
+	 */
+
+	while ((opt = getopt(argc, argv, ":a:r:c")) != ERROR) {
+		switch (opt) {
+			case 'a': 
+				/* -a: Select alarm id, if alarm is set, then it is
+				 * done by absolute time (RTC_SET_ALARM ioctl call).
+				 */
+				alarmid = strtol(optarg, NULL, 0);
+				if (alarmid < 0 || alarmid >= CONFIG_RTC_NALARMS) { 
+					badarg = true;
+				}
+				break;
+			case 'r': 
+				/* -r: Select alarm id, if alarm is set,then it is
+				 *  done by relative time (RTC_SET_RELATIVE ioctl call). 
+				 */
+				alarmid = strtol(optarg, NULL, 0);
+				if (alarmid < 0 || alarmid >= CONFIG_RTC_NALARMS) { 
+					badarg = true;
+				}
+				relset = true;
+				break;
+			case 'c':
+				cancelmode = true;
+				break;
+			default:
+				fprintf(stderr, "<unknown parameter '-%c'>\n\n", opt);
+				/* fall through */
+
+			case '?':
+			case ':':
+				badarg = true;
+		}
+	}
+
+	setmode = !cancelmode;
```

Comment:
This variable is not required. If its not cancel mode, then its set mode always.